### PR TITLE
Add Interactive brokered authentication and wam content

### PR DESCRIPTION
**Pull request by alexwolfmsft**
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/azure/sdk/authentication/additional-methods.md](https://github.com/dotnet/docs/blob/50125118a866d31f50e93d488d1db6a143273815/docs/azure/sdk/authentication/additional-methods.md) | [docs/azure/sdk/authentication/additional-methods](https://review.learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/additional-methods?branch=pr-en-us-42279) |

### CODE REVIEW

File: docs/azure/sdk/authentication/additional-methods.md [hunk 1]
```diff
@@ -36,6 +36,64 @@ foreach (var blobItem in client.GetBlobContainers())
 
 For more exact control, such as setting redirect URIs, you can supply specific arguments to `InteractiveBrowserCredential` such as `redirect_uri`.
 
+### Authenticate the default system account via WAM
```
Comment (scottaddie) [inline docs/azure/sdk/authentication/additional-methods.md:39]: Let's stick with the same heading Python has.
```suggestion
## Interactive brokered authentication
```

File: docs/azure/sdk/authentication/additional-methods.md [hunk 2]
```diff
@@ -36,6 +36,64 @@ foreach (var blobItem in client.GetBlobContainers())
...
+
+Web Account Manager (WAM) is a system authentication broker service that allows apps to seamlessly request OAuth tokens from identity providers, such as Microsoft Entra ID. WAM enables identity providers to natively plug into the OS and provide the service to other apps to streamline the login process. WAM offers the following benefits:
```
Comment (scottaddie) [inline docs/azure/sdk/authentication/additional-methods.md:41]: It's important to call out right away that this feature is for Windows only. We don't yet have a solution for macOS and Linux.
```suggestion
Web Account Manager (WAM) is a system authentication broker service for Windows that allows apps to seamlessly request OAuth tokens from identity providers, such as Microsoft Entra ID. WAM enables identity providers to natively plug into the OS and provide the service to other apps to streamline the login process. WAM offers the following benefits:
```

File: docs/azure/sdk/authentication/additional-methods.md [hunk 3]
```diff
@@ -36,6 +36,64 @@ foreach (var blobItem in client.GetBlobContainers())
...
+
+- **Feature support**: Apps can access OS-level and service-level capabilities, including Windows Hello, conditional access policies, and FIDO keys.
+- **Streamlined single sign-on**: Apps can use the built-in account picker, allowing the user to select an existing account instead of repeatedly entering the same credentials.
+- **Enhanced security**: Bug fixes and enhancements ship with Windows.
+- **Token protection**: Refresh tokens are device-bound, and apps can acquire device-bound access tokens.
+
+Many people always sign in to Windows with the same user account and, therefore, only ever want to authenticate using that account. WAM also supports a silent login process that automatically uses a default account so the user does not have to repeatedly select it.
+
+To use WAM and the default system account in your app:
```
Comment (scottaddie) [inline docs/azure/sdk/authentication/additional-methods.md:50]:
```suggestion
To use WAM in your app:
```

File: docs/azure/sdk/authentication/additional-methods.md [hunk 4]
```diff
@@ -36,6 +36,64 @@ foreach (var blobItem in client.GetBlobContainers())
...
+                // Enable silent brokered authentication using the default account
+                UseDefaultBrokerAccount = true,
+            }
+        );
+    }
+    ```
+
+1. Use the broker-enabled `InteractiveBrowserCredential` instance.
+
+Once you opt into this behavior, the credential type attempts to sign in by asking the underlying Microsoft Authentication Library (MSAL) to perform the sign-in for the default system account. If the sign-in fails, the credential type falls back to displaying the account picker dialog, from which the user can select the appropriate account.
```
Comment (scottaddie) [inline docs/azure/sdk/authentication/additional-methods.md:95]:
```suggestion
Once you opt into this behavior, the credential attempts to sign in by asking the underlying Microsoft Authentication Library (MSAL) to perform the sign-in for the default system account. If the sign-in fails, the credential type falls back to displaying the account picker dialog, from which the user can select the appropriate account.
```

File: docs/azure/sdk/authentication/additional-methods.md [hunk 5]
```diff
@@ -36,6 +36,64 @@ foreach (var blobItem in client.GetBlobContainers())
...
+    private async void testBrokeredAuth_Click(object sender, EventArgs e)
+    {
+        IntPtr windowHandle = this.Handle;
+    }
+    ```
+
+    > [!NOTE]
+    > This example shows how to get the window handle for a Windows Forms app. Visit the [Parent window handles](/entra/msal/dotnet/acquiring-tokens/desktop-mobile/wam#parent-window-handles) and [Retrieve a window handle](/windows/apps/develop/ui-input/retrieve-hwnd) articles for more information about retrieving window context for other types of apps.
+
+1. Create an instance of `InteractiveBrowserCredential` in your app. The credential requires the handle of the parent window that's requesting the authentication flow. On Windows, the handle is an integer value that uniquely identifies the window. Optionally, set the `UseDefaultBrokerAccount` option to `true` to enable silent brokered authentication, which will automatically select the default account.
```
Comment (scottaddie) [inline docs/azure/sdk/authentication/additional-methods.md:72]:
```suggestion
1. Create an instance of `InteractiveBrowserCredential` in your app. The credential requires the handle of the parent window that's requesting the authentication flow. On Windows, the handle is an integer value that uniquely identifies the window. Optionally, set the [UseDefaultBrokerAccount](/dotnet/api/azure.identity.broker.interactivebrowsercredentialbrokeroptions.usedefaultbrokeraccount#azure-identity-broker-interactivebrowsercredentialbrokeroptions-usedefaultbrokeraccount) property to `true` to enable silent, brokered authentication, which will automatically select the default system account.
```

**Review by scottaddie — CHANGES_REQUESTED** [review]
These set up instructions need to be included somewhere too: https://learn.microsoft.com/en-us/azure/developer/python/sdk/authentication-additional-methods#enable-applications-for-interactive-brokered-authentication

File: docs/azure/sdk/authentication/additional-methods.md [hunk 6]
```diff
@@ -36,6 +51,122 @@ foreach (var blobItem in client.GetBlobContainers())
...
+
+// App code omitted for brevity
+
+IntPtr windowHandle = this.Handle;
+
+var credential = new InteractiveBrowserCredential(
+    new InteractiveBrowserCredentialBrokerOptions(windowHandle)
+    {
+        // Enable silent brokered authentication using the default account
+        UseDefaultBrokerAccount = true,
```
Comment (scottaddie) [inline docs/azure/sdk/authentication/additional-methods.md:163]: I recommend creating a sample app to accompany this doc. Then you can import code snippets and highlight important lines of code like this one.

File: docs/azure/sdk/authentication/additional-methods.md [hunk 7]
```diff
@@ -16,6 +16,21 @@ This method interactively authenticates an application through [`InteractiveBrow
...
+1. On the [Azure portal](https://portal.azure.com), navigate to Microsoft Entra ID and select **App registrations** on the left navigation.
+1. Select the registration for your app, then select **Authentication**.
+1. Under **Advanced settings**, select **Yes** for **Allow public client flows**.
+1. Select **Save** to apply the changes.
+1. To authorize the application for specific resources, navigate to the resource in question, select **API Permissions**, and enable **Microsoft Graph** and other resources you want to access. Microsoft Graph is usually enabled by default.
+
+    > [!IMPORTANT]
+    > You must also be the admin of your tenant to grant consent to your application when you sign in for the first time.
+
+If you can't configure the device code flow option, your application might need to be multitenant. To make this change, navigate to the **Authentication** panel, select **Accounts in any organizational directory** (under **Supported account types**), and then select **Yes** for **Allow public client flows**.
```
Comment (christothes) [inline docs/azure/sdk/authentication/additional-methods.md:32]: This paragraph about device code flow option seems out of place - to which device code option are we referring?

File: docs/azure/sdk/authentication/additional-methods.md [hunk 8]
```diff
@@ -36,6 +51,122 @@ foreach (var blobItem in client.GetBlobContainers())
 
 For more exact control, such as setting redirect URIs, you can supply specific arguments to `InteractiveBrowserCredential` such as `redirect_uri`.
 
+## Interactive brokered authentication
+
+This method interactively authenticates an application through [`InteractiveBrowserCredential`](/dotnet/api/azure.identity.interactivebrowsercredential?view=azure-dotnet) by collecting user credentials using the system authentication broker. A system authentication broker is an app running on a user's machine that manages the authentication handshakes and token maintenance for all connected accounts. Currently, only the Windows authentication broker, Web Account Manager (WAM), is supported. Users on macOS and Linux will be authenticated through a browser.
```
Comment (christothes) [inline docs/azure/sdk/authentication/additional-methods.md:56]:
```suggestion
This method interactively authenticates an application through [`InteractiveBrowserCredential`](/dotnet/api/azure.identity.interactivebrowsercredential?view=azure-dotnet) by collecting user credentials using the system authentication broker. A system authentication broker is an app running on a user's machine that manages the authentication handshakes and token maintenance for all connected accounts. Currently, only the Windows authentication broker, Web Account Manager (WAM), is supported. Users on macOS and Linux will be authenticated through the non-brokered interactive browser flow.
```

File: docs/azure/sdk/authentication/additional-methods.md [hunk 9]
```diff
@@ -36,6 +51,122 @@ foreach (var blobItem in client.GetBlobContainers())
...
+
+WAM enables identity providers such as Microsoft Entra ID to natively plug into the OS and provide the service to other apps to streamline the login process. WAM offers the following benefits:
```
Comment (christothes) [inline docs/azure/sdk/authentication/additional-methods.md:58]:
```suggestion
WAM enables identity providers such as Microsoft Entra ID to natively plug into the OS and provide the service to other apps to provide a more secure login process. WAM offers the following benefits:
```

File: docs/azure/sdk/authentication/additional-methods.md [hunk 10]
```diff
@@ -36,6 +51,122 @@ foreach (var blobItem in client.GetBlobContainers())
...
+
+- **Feature support**: Apps can access OS-level and service-level capabilities, including Windows Hello, conditional access policies, and FIDO keys.
+- **Streamlined single sign-on**: Apps can use the built-in account picker, allowing the user to select an existing account instead of repeatedly entering the same credentials.
+- **Enhanced security**: Bug fixes and enhancements ship with Windows.
+- **Token protection**: Refresh tokens are device-bound, and apps can acquire device-bound access tokens.
+
+Personal Microsoft accounts and work or school accounts are supported. If a supported version of Windows is used, the default browser-based UI is replaced with a smoother authentication experience, similar to Windows built-in apps.
+
+Interactive brokered authentication enables the application for all operations allowed by the interactive login credentials. As a result, if you're the owner or administrator of your subscription, your code has inherent access to most resources in that subscription without having to assign any specific permissions.
```
Comment (christothes) [inline docs/azure/sdk/authentication/additional-methods.md:67]: Why is this special to brokered auth?

File: docs/azure/sdk/authentication/additional-methods.md (hunk 7 above)
Comment (alexwolfmsft) [reply docs/azure/sdk/authentication/additional-methods.md]: Removed

**Review by scottaddie — APPROVED** [review]

File: docs/azure/sdk/snippets/additional-auth/interactive/InteractiveBrokeredAuth.cs [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+        private void testInteractiveBrokeredAuth_Click(object sender, EventArgs e)
+        {
+            // Get the handle of the current window
+            IntPtr windowHandle = this.Handle;
+
+            var credential = new InteractiveBrowserCredential(
+                new InteractiveBrowserCredentialBrokerOptions(windowHandle));
+
+            // To authenticate and authorize with an app, substitute the
```
[the remark this answers is not in the document]
Comment (scottaddie) [reply docs/azure/sdk/snippets/additional-auth/interactive/InteractiveBrokeredAuth.cs:22]: Ship it!

**Review by christothes — APPROVED** [review]

**Pull request merged**